editor/block-switcher/index.js

```diff
@@ -110,9 +110,7 @@ function BlockSwitcher( { blocks, onTransform } ) {
 								} }
 								className="editor-block-switcher__menu-item"
 								icon={ (
-									<span className="editor-block-switcher__block-icon">
```

Comment:
The block icon is not always a dashicon, this wrapper was necessary to correctly add the spacing between the icon and text when we have custom elements as icons. Not saying we shouldn't remove it but it was there on purpose and we should make sure it doesn't break. #3238 

I think you're targetting `svg` but block icons are not always `svg`, they can be anything.